pkg/cliplugins/bind/plugin/bind_compute.go

```diff
@@ -0,0 +1,320 @@
+		return err
+	}
+
+	if err := o.applyPlacement(ctx, userWorkspaceKcpClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+const maxBindingNamePrefixLength = validation.DNS1123SubdomainMaxLength - 1 - 8
+
+func apiBindingName(clusterName logicalcluster.Name, apiExportName string) string {
+	maxLen := len(apiExportName)
+	if maxLen > maxBindingNamePrefixLength {
+		maxLen = maxBindingNamePrefixLength
+	}
+	bindingNamePrefix := apiExportName[:maxLen]
+
+	hash := sha256.Sum224([]byte(clusterName.Path()))
+	base36hash := strings.ToLower(base36.EncodeBytes(hash[:]))
+	return fmt.Sprintf("%s-%s", bindingNamePrefix, base36hash[:8])
+}
+
+func (o *BindComputeOptions) applyAPIBinding(ctx context.Context, client kcpclient.Interface, desiredAPIExports sets.String) error {
+	apiBindings, err := client.ApisV1alpha1().APIBindings().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	existingAPIExports := sets.NewString()
+	for _, binding := range apiBindings.Items {
+		if binding.Spec.Reference.Workspace == nil {
+			continue
+		}
+		existingAPIExports.Insert(fmt.Sprintf("%s:%s", binding.Spec.Reference.Workspace.Path, binding.Spec.Reference.Workspace.ExportName))
+	}
+
+	diff := desiredAPIExports.Difference(existingAPIExports)
+	var errs []error
+	for export := range diff {
+		clusterName, name := logicalcluster.New(export).Split()
+		apiBinding := &apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: apiBindingName(clusterName, name),
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				Reference: apisv1alpha1.ExportReference{
+					Workspace: &apisv1alpha1.WorkspaceExportReference{
+						Path:       clusterName.String(),
+						ExportName: name,
+					},
+				},
+			},
+		}
+		_, err := client.ApisV1alpha1().APIBindings().Create(ctx, apiBinding, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			errs = append(errs, err)
+		}
+
+		_, err = fmt.Fprintf(o.Out, "apibinding %s for apiexport %s created.\n", apiBinding.Name, export)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (o *BindComputeOptions) applyPlacement(ctx context.Context, client kcpclient.Interface) error {
+	placement := &schedulingv1alpha1.Placement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: o.PlacementName,
+		},
+		Spec: schedulingv1alpha1.PlacementSpec{
+			NamespaceSelector: o.namespaceSelector,
+			LocationSelectors: o.locationSelectors,
+			LocationWorkspace: o.LocationWorkspace.String(),
+			LocationResource: schedulingv1alpha1.GroupVersionResource{
+				Group:    "workload.kcp.dev",
+				Version:  "v1alpha1",
+				Resource: "synctargets",
+			},
+		},
+	}
+
+	placement, err := client.SchedulingV1alpha1().Placements().Create(ctx, placement, metav1.CreateOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	// wait for placement to be ready
+	conditions.Get(placement, schedulingv1alpha1.PlacementReady)
```

Comment:
I assume we would also want to wait for the newly-created bindings to be successfully bound.
Maybe we can do this wait for both placement and bindings in a dedicated function ?